transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -354,14 +359,20 @@ private void releaseAndOffer(Channel channel, Promise<Void> promise) throws Exce
         }
     }
 
-    private void closeChannel(Channel channel) {
+    private void closeChannel(Channel channel) throws Exception {
         channel.attr(POOL_KEY).getAndSet(null);
         channel.close();
     }
 
     private void closeAndFail(Channel channel, Throwable cause, Promise<?> promise) {
-        closeChannel(channel);
-        promise.tryFailure(cause);
+        try {
+            if (channel != null) {
+                closeChannel(channel);
+            }
+            promise.tryFailure(cause);
+        } catch (Throwable t) {
```

Comment:
In this case it may be better to just have the `try-catch` around `closeChannel()` as retrying `tryFailure()` if it fails is futile.

Reply:
Done

Reply:
Done